src/Devices/Kiosk/KioskDevice.cs

```diff
@@ -84,7 +86,7 @@ private Task<MethodResponse> ReceivePurchaseTicketResponse(MethodRequest methodR
             string result = "{\"result\":\"Executed direct method: " + methodRequest.Name + "\"}";
 
             // restart timer
-            this.eventScheduler.StartAll(); // TODO: need more elegant solution
+            this.purchaseticketEvent.Start(); // TODO: need more elegant solution
```

Comment:
This removes the event scheduler abstraction. This will again interfere in the testing. I wonder why the change is needed to begin with, given that the Kiosk device owns the scheduler now, and it knows that it only has purchase ticket event.

Reply:
the tests actually still function without error. The primary reason is that if the class is enhanced in the future with additional timed events (like a simulated device failure), we don't restart everything. We really only want to restart the one device. 

The observer pattern would have been idea for this. But we need to get this in ASAP as we're already weeks behind schedule. So I'm opting or less elegant solution.

Reply:
When the future is there, we can add the capability on the scheduler to manage a specific event rather than all. For now, this does interfere with the testing (just didn't have the chance yet -- and in very high CPU test runs, it very well might have that chance), and it also is not verifiable, in the sense that our test cannot verify now that starts been called by the callback. Please change it back for now, and if you have time add validation in the test that start all on the scheduler has been called

Reply:
"start all" has been called will require some effort as the fakeScheduler doesn't track a start/stop value for the timer. I'll get both in shortly.

Reply:
updated. FakeSimulatedEvent will now track and report a simulated running state.